integration/simulation/transaction_injector.go

```diff
@@ -347,12 +361,19 @@ func readNonce(cl *obscuroclient.Client, a common.Address) uint64 {
 }
 
 func NextNonce(cl *obscuroclient.Client, w wallet.Wallet) uint64 {
+	counter := 0
+
 	// only returns the nonce when the previous transaction was recorded
 	for {
 		result := readNonce(cl, w.Address())
 		if result == w.GetNonce() {
 			return w.GetNonceAndIncrement()
 		}
 		time.Sleep(time.Millisecond)
+		counter++
+
+		if counter > timeoutMillis {
+			panic("Transaction injector failed to retrieve nonce after ten seconds...")
```

Comment:
This used to spin forever if for some reason the nonce wasn't increasing (e.g. the enclave was down).

Reply:
Hitting it a thousand times a second feels a bit intense for 10 seconds. If the server is struggling under load then this is the sort of thing that causes a death spiral.

If it normally takes a couple of millis you could back off, like `time.sleep(2^retries * time.Millisecond)` and then `if (retries > 10) { panic(...) }`